source/common/filesystem/filesystem_impl.cc

```diff
@@ -160,9 +160,9 @@ void FileImpl::flushThreadFunc() {
       try {
         if (reopen_file_) {
           reopen_file_ = false;
-          const auto result = raw_file_->close();
+          const Api::SysCallBoolResult result = raw_file_->close();
           ASSERT(result.rc_, fmt::format("unable to close file '{}': {}", raw_file_->path(),
-                                         strerror(result.errno_)));
+                                         raw_file_->errorToString(result.errno_)));
```

Comment:
Better, but I feel like even the errno_ could be abstracted here. Can we have errorString() be a method on Api::SysCallBoolResult, or something like that?

E.g. I was envisioning on Windows or some other hypothetical OS that an error might have a different representation than an 'int' for errors. It might be an enum or some other abstract type. WDYT?

We could also do this on a follow-up.

Reply:
I think for now we can hold off on addressing this, since our implementation of RawFile on Windows will also use errno + strerror as well.

This will definitely come up when we start PRing socket changes upstream (e.g. an implementation of https://github.com/envoyproxy/envoy/pull/5829). In Windows sockets, errors are not reported in errno and are not displayed using strerror.  With the full context, I think will be able to come up with the correct abstraction

Reply:
OK can you add a TODO for that?

Thanks!

Reply:
we can add the TODO